djcloudbridge/view_helpers.py

```diff
@@ -188,6 +188,11 @@ def get_credentials_by_id(cloud, request, credentials_id):
     """
     if request.user.is_anonymous:
         return {}
+
+    if not hasattr(request.user, 'userprofile'):
```

Comment:
Since this is a get method, I think that ideally, it shouldn't mutate state, as that would be doing something other than what is advertised. The recommended way I've seen to handle this profile situation in Django is to use signals. There's an example for that exact case here: https://simpleisbetterthancomplex.com/tutorial/2016/07/28/how-to-create-django-signals.html

Reply:
What if, in order to not have the error from trying to access the userprofile and not have any other issues, we just return the empty dict if there is no userprofile (implying there are no credentials), and never create the empty userprofile silently, but rather wait for the user to create it as we are doing now?
My thought is that we don't really need to create a blank profile as it's not necessary to use any appliance, and it would also avoid us naming userprofiles by default on behalf of users, and it would also make the bug appearing in sentry disappear?